cdap-security/src/main/java/co/cask/cdap/security/securestore/FileSecureStoreMetadata.java

```diff
@@ -0,0 +1,188 @@
+import org.apache.commons.io.Charsets;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents the metadata for the data stored in the Secure Store.
+ */
+class FileSecureStoreMetadata implements SecureStoreMetadata {
+
+  private static final String NAME_FIELD = "name";
+  private static final String DESCRIPTION_FIELD = "description";
+  private static final String CREATED_FIELD = "created";
+  private static final String PROPERTIES_FIELD = "properties";
+  private static final String DESCRIPTION_DEFAULT = "";
+
+  private final String name;
+  private final String description;
+  private final Date created;
+  private final Map<String, String> properties;
+
+  private FileSecureStoreMetadata(String name, String description, Date created, Map<String, String> properties) {
+    this.name = name;
+    this.description = description;
+    this.created = created;
+    this.properties = properties;
+  }
+
+  public static SecureStoreMetadata of(String name, Map<String, String> properties) {
+    String tempDescription = properties.get(DESCRIPTION_FIELD) == null ?
+      DESCRIPTION_DEFAULT : properties.get(DESCRIPTION_FIELD);
+
+    return new FileSecureStoreMetadata(name, tempDescription, new Date(), properties);
+  }
+
+  /**
+   * @return Name of the data.
+   */
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * @return Last time, in epoch, this element was modified.
+   */
+  @Override
+  public long getLastModifiedTime() {
+    return created.getTime();
+  }
+
+  /**
+   * @return A map of properties associated with this element.
+   */
+  @Override
+  public Map<String, String> getProperties() {
+    return properties;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public Date getCreated() {
+    return created;
+  }
+
+  @Override
+  public String toString() {
+    final StringBuilder metaSB = new StringBuilder();
+    metaSB.append("name: ").append(name).append(", ");
+    metaSB.append("description: ").append(description).append(", ");
+    metaSB.append("created: ").append(created).append(", ");
+    metaSB.append("properties: ");
+    if ((properties != null) && !properties.isEmpty()) {
+      for (Map.Entry<String, String> property : properties.entrySet()) {
+        metaSB.append("[");
+        metaSB.append(property.getKey());
+        metaSB.append("=");
+        metaSB.append(property.getValue());
+        metaSB.append("], ");
+      }
+      metaSB.deleteCharAt(metaSB.length() - 2);  // remove last ', '
+    } else {
+      metaSB.append("null");
+    }
+    return metaSB.toString();
+  }
+
+  /**
+   * Serialize the metadata to a byte array.
+   *
+   * @return the serialized bytes
+   * @throws IOException
+   */
+  byte[] serialize() throws IOException {
```

Comment:
Some thing like 

`ByteArrayOutputStream buffer = new ByteArrayOutputStream();
 OutputStreamWriter out = new OutputStreamWriter(buffer, Charsets.UTF_8);
 GSON.toJson(this, out);
 return buffer.toByteArray();`